File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/test/TestSwlTtlKeyExchanger.java
package i2f.swl.test;

import i2f.jce.std.encrypt.asymmetric.key.AsymKeyPair;
import i2f.swl.core.key.ttl.SwlTtlKeyExchanger;
import i2f.swl.data.SwlData;
import i2f.swl.impl.SwlAesSymmetricEncryptor;
import i2f.swl.impl.SwlBase64Obfuscator;
import i2f.swl.impl.SwlRsaAsymmetricEncryptor;
import i2f.swl.impl.SwlSha256MessageDigester;

import java.util.Arrays;

/**
 * @author Ice2Faith
 * @date 2024/7/10 10:12
 * @desc
 */
public class TestSwlTtlKeyExchanger {
    public static void main(String[] args) {

        SwlTtlKeyExchanger clientTransfer = new SwlTtlKeyExchanger();
        clientTransfer.setAsymmetricEncryptorSupplier(() -> new SwlRsaAsymmetricEncryptor());
        clientTransfer.setSymmetricEncryptorSupplier(() -> new SwlAesSymmetricEncryptor());
        clientTransfer.setMessageDigester(new SwlSha256MessageDigester());
        clientTransfer.setObfuscator(new SwlBase64Obfuscator());

        SwlTtlKeyExchanger serverTransfer = new SwlTtlKeyExchanger();
        serverTransfer.setAsymmetricEncryptorSupplier(() -> new SwlRsaAsymmetricEncryptor());
        serverTransfer.setSymmetricEncryptorSupplier(() -> new SwlAesSymmetricEncryptor());
        serverTransfer.setMessageDigester(new SwlSha256MessageDigester());
        serverTransfer.setObfuscator(new SwlBase64Obfuscator());

        AsymKeyPair serverKeyPair = serverTransfer.resetSelfKeyPair();
        AsymKeyPair clientKeyPair = clientTransfer.resetSelfKeyPair();

        clientTransfer.acceptOtherPublicKey(serverKeyPair.getPublicKey());
        serverTransfer.acceptOtherPublicKey(clientKeyPair.getPublicKey());

        SwlData clientSendData = clientTransfer.sendByKey(serverKeyPair.getPublicKey(), Arrays.asList("body:123456", "query:user=admin"));


        String clientId = "127.0.0.1";
        SwlData serverReceiveData = serverTransfer.receiveByKey(clientId, clientSendData);

        SwlData serverResponseData = serverTransfer.responseByKey(serverReceiveData.getHeader().getRemoteAsymSign(), Arrays.asList("echo:ok", "data:ok"));

        String serverId = "server";
        SwlData clientReceiveData = clientTransfer.receiveByKey(serverId, serverResponseData);

        System.out.println("ok");
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/test/TestSwlTtlKeyExchanger.js
import SwlRsaAsymmetricEncryptorSupplier from "../core/impl/supplier/SwlRsaAsymmetricEncryptorSupplier";
import SwlAesSymmetricEncryptorSupplier from "../core/impl/supplier/SwlAesSymmetricEncryptorSupplier";
import SwlSha256MessageDigester from "../core/impl/SwlSha256MessageDigester";
import SwlBase64Obfuscator from "../core/impl/SwlBase64Obfuscator";
import SwlTtlKeyExchanger from "../core/core/key/ttl/SwlTtlKeyExchanger";
import SwlMemTtlKeyManager from "../core/core/key/ttl/impl/SwlMemTtlKeyManager";

/**
 * @return TestSwlTtlKeyExchanger
 * @constructor
 */
function TestSwlTtlKeyExchanger() {

}

TestSwlTtlKeyExchanger.main = function () {

    let clientTransfer = new SwlTtlKeyExchanger();
    clientTransfer.asymmetricEncryptorSupplier = (new SwlRsaAsymmetricEncryptorSupplier());
    clientTransfer.symmetricEncryptorSupplier = (new SwlAesSymmetricEncryptorSupplier());
    clientTransfer.messageDigester = (new SwlSha256MessageDigester());
    clientTransfer.obfuscator = (new SwlBase64Obfuscator());
    clientTransfer.keyManager=new SwlMemTtlKeyManager()

    let serverTransfer = new SwlTtlKeyExchanger();
    serverTransfer.asymmetricEncryptorSupplier = (new SwlRsaAsymmetricEncryptorSupplier());
    serverTransfer.symmetricEncryptorSupplier = (new SwlAesSymmetricEncryptorSupplier());
    serverTransfer.messageDigester = (new SwlSha256MessageDigester());
    serverTransfer.obfuscator = (new SwlBase64Obfuscator());
    serverTransfer.keyManager=new SwlMemTtlKeyManager()

    let serverKeyPair = serverTransfer.resetSelfKeyPair();
    let clientKeyPair = clientTransfer.resetSelfKeyPair();

    clientTransfer.acceptOtherPublicKey(serverKeyPair.getPublicKey());
    serverTransfer.acceptOtherPublicKey(clientKeyPair.getPublicKey());

    let clientSendData = clientTransfer.sendByKey(serverKeyPair.getPublicKey(), ["body:123456", "query:user=admin"]);


    let clientId = "127.0.0.1";
    let serverReceiveData = serverTransfer.receiveByKey(clientId, clientSendData);

    let serverResponseData = serverTransfer.responseByKey(serverReceiveData.header.remoteAsymSign, ["echo:ok", "data:ok"]);

    let serverId = "server";
    let clientReceiveData = clientTransfer.receiveByKey(serverId, serverResponseData);

    console.log("ok");
}

export default TestSwlTtlKeyExchanger
